codecov_cli/commands/upload.py

```diff
@@ -17,169 +17,183 @@ def _turn_env_vars_into_dict(ctx, params, value):
     return dict((v, os.getenv(v, None)) for v in value)
 
 
+_global_upload_options = [
```

Comment:
This is quite nice. I think however that we could break this up even more. My suggestion would be to have a `base_cli_command_option` possibly defined in `commands/__init__.py` or some helper file. These could include the `--sha`, `--token`, `--dry-run`, `--fail-on-error` options.

Perhaps this is better suited for another round of changes if we decide to do it, so I'll leave just as a comment. What do you think? I guess we'd have to add support for some of these options to different commands...

Reply:
sounds like a reasonable change. I'll make a ticket for this